### ISSUE TITLE
Feature Request: add support for external registries as sources and destinations

### ISSUE THREAD

**Issue by scanlonp** [cdklabs/cdk-docker-image-deployment#10]
Depends on #7 and #8. #9 is an example. Other examples are ECR and Artifactory.

- [ ] External ECR repositories
- [ ] DockerHub
- [ ] Aritfactory

Add to this list and make a new issue if there are other registries you would like to be supported.

**Comment by bcosynot** [comment]
I am curious how this compares to https://github.com/cdklabs/cdk-ecr-deployment ?
Can move to another issue if its off-topic, lmk.

**Comment by Richardmbs12** [comment]
cdk-ecr-deployment doesnt support java :(
